### DISCUSSION TITLE
CoreWCF Server on .NET Framework without WCF libraries

### DISCUSSION THREAD

**Discussion by denisbredikhin** [CoreWCF/CoreWCF#540] [Q&A]
Hello,

we are migrating complex enterprise-grade application to .NET Core (.NET 6) and we see migration to CoreWCF as one of the first steps.
To achieve this we need to use CoreWCF in .NET Framework application without the use of ServiceHost model. Then we can migrate this code easily to .NET Core when other parts of the system will be ready.
Is is possible to achieve this? I read everywhere that CoreWCF is based on ASP.NET Core. I know that in the past there was such thing as ASP.NET Core for .NET Framework, but I'm not sure if it is supported by MS and if it is supported by CoreWCF.

**Comment by mconnew** [comment]
Yes you can do that. CoreWCF is specifically supported on .NET Framework for exactly this reason. ASP.NET Core 2.1 is still supported on .NET Framework (but not .NET Core any longer). We still run all our tests on .NET Framework with ASP.NET Core 2.1 as well as .NET Core 3.1 through to .NET 6. We are soon approaching releasing the 1.0 release (next few months), after which most feature development will freeze for .NET Framework as the 2.0 version of CoreWCF will have a dependency on .NET 6.
